File: backAnimalCrossing/src/main/java/com/ironhack/backAnimalCrossing/model/FishCatched.java
package com.ironhack.backAnimalCrossing.model;

import javax.persistence.*;

@Entity
public class FishCatched {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToOne
    @JoinColumn(name = "islandLeader_id")
    private IslandLeader islandLeader;

    private String[] fishes;

    public FishCatched() {
    }

    public FishCatched(String name, IslandLeader islandLeader, String[] fishes) {
        this.name = name;
        this.islandLeader = islandLeader;
        this.fishes = fishes;
    }

    public FishCatched(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }



    public String[] getFishes() {
        return fishes;
    }

    public void setFishes(String[] fishes) {
        this.fishes = fishes;
    }

    public IslandLeader getIslandLeader() {
        return islandLeader;
    }

    public void setIslandLeader(IslandLeader islandLeader) {
        this.islandLeader = islandLeader;
    }
}

File: frontAnimalCrossing/src/app/models/fishCatched.ts
import { IslandLeader } from "./islandLeader";

export class FishCatched{

  constructor(
  private _id: number | null,
  private _name: string,
  private _islandLeader: IslandLeader,
  private _fishCatched: string[]
  ){}

  public get fishCatched(): string[] {
    return this._fishCatched;
  }
  public set fishCatched(value: string[]) {
    this._fishCatched = value;
  }

  public get islandLeader(): IslandLeader {
    return this._islandLeader;
  }
  public set islandLeader(value: IslandLeader) {
    this._islandLeader = value;
  }
  public get name(): string {
    return this._name;
  }
  public set name(value: string) {
    this._name = value;
  }
  public get id(): number | null {
    return this._id;
  }
  public set id(value: number | null) {
    this._id = value;
  }

  public toJSON(): any {
    return {
      id: this._id,
      name: this._name,
      islandLeader: this._islandLeader,
      fishCatched: this._fishCatched
    };
  }
}
